### PR TITLE
Remove SOS and lldb plugin

### PR DESCRIPTION
**Pull request by mikem8361**
Issue: https://github.com/dotnet/coreclr/issues/24092

### CODE REVIEW

**Comment by danmosemsft** [comment]
@stephentoub

File: Documentation/building/debugging-instructions.md [hunk 1]
```diff
@@ -24,29 +26,25 @@ Steps 1-8 only need to be done once, and then (9) can be repeated whenever you w
...
-To use lldb on OS X, you first need to build it and the SOS plugin on the machine you intend to use it. See the instructions in [building lldb](buildinglldb.md). The rest of instructions on how to use lldb for Linux on are the same.
+Debugging CoreCLR on Linux and macOS
+====================================
 
-Debugging CoreCLR on Linux
-==========================
+See the SOS installation instructions [here](https://github.com/dotnet/diagnostics/blob/master/documentation/installing-sos-windows-instructions.md). After SOS is installed, it will automatically be loaded by lldb.
 
-Only lldb is supported by the SOS plugin. gdb can be used to debug the coreclr code but with no SOS support. Visual Studio 2015 RTM remote debugging isn't currently supported.
+Only lldb is supported by the SOS. gdb can be used to debug the coreclr code but with no SOS support. Visual Studio 2015 RTM remote debugging isn't currently supported.
```
Comment (noahfalk) [inline Documentation/building/debugging-instructions.md]:
```suggestion
Only lldb is supported by SOS. Gdb can be used to debug the coreclr code but with no SOS support. Visual Studio 2015 RTM remote debugging isn't currently supported.
```

**Comment by AndyAyersMS** [comment]
@mikem8361 you probably mentioned this somewhere, but what's the best way to get an SOS now that it no longer gets built as part of CoreCLR? The diagnostics repo does not seem to point us to downloads, so does that mean I have to clone the repo and build it myself?

Any chance we could get some kind of package restore path so that at least in the Core_Root directory the tools that use to get built and end up there are still present?

**Comment by mikem8361** [comment]
SOS can be installed see: https://github.com/dotnet/diagnostics#installing-sos

On Windows with a newer version of Windbg, the new SOS will be automatically downloaded and loaded from the internal extension gallery.

**Comment by mikem8361** [comment]
Let me know if you have any problems or questions.

**Comment by AndyAyersMS** [comment]
I was on linux arm64 and ended up building it myself, but using `dotnet` to fetch it and set up an entry in .lldbinit looks useful -- will try that in the future.

**Pull request merged**